### PR TITLE
Client and Browser Integration!!

### PR DESCRIPTION
**Pull request by Enriikke**
- Tied up the client and browser models to work together
- Added support for application_name and application version definitions
- Refactored browser object to use Skoogle Docs documents
- Added support to include SkoogleDocs::Document tests: cassette and factory
- Cleaned up documentation
- Other minor changes: removed yardstick from guard, fixed guard to run spec on save, etc.

### CODE REVIEW

**Comment by iamdustan** [comment]
I’d love if I could get someone way more qualified than me to be reviewing this. @ruprict?

Also, @patlaughlin to at least give you another venue to be reading ruby

**Comment by patlaughlin** [comment]
@iamdustan awesome yeah. I'll be keeping an eye on this repo. Potentially writing asinine code reviews

File: lib/skoogle_docs/client.rb [hunk 1]
```diff
@@ -117,14 +154,5 @@ def redirect_uri
     def session
```
Comment (ruprict) [inline lib/skoogle_docs/client.rb]: So, the browser is initialized with the session, why do we need it in the client? It seems having it in both places could lead to strangeness.

File: lib/skoogle_docs/client.rb [hunk 2]
```diff
@@ -47,13 +58,30 @@ def initialize(options = {})
       yield(self) if block_given?
```
Comment (ruprict) [inline lib/skoogle_docs/client.rb]: Should we consider a Configuration object?

File: lib/skoogle_docs/session.rb [hunk 3]
```diff
@@ -27,10 +27,6 @@ def initialize(client)
     # Wrapper to the Google Drive API
     #
     # @return [Google::APIClient::API]
-    #
-    # @example Accessing Google Drive API
-    #   session = SkoogleDocs::Session.new(client)
-    #   session.api.files.get
     def api
       @drive
```
Comment (ruprict) [inline lib/skoogle_docs/session.rb]: What if this is nil, meaning, the auth failed? Also, should we validate the params in `execute`? Do we know what errors the Google API can raise?

**Comment by ruprict** [comment]
I did an initial review. I'm sure @Enriikke will want to chat about it. Looks really good, for the most part (except the build failing bit....what up? ;) JRuby ruins everything....)

File: lib/skoogle_docs/session.rb (hunk 3 above)
Comment (Enriikke) [inline lib/skoogle_docs/session.rb]: We talked over this with @ruprict. If auth fails then you don't even get to this api call. I agree with Glenn that validating params in the execute call is important.

**Comment by ruprict** [comment]
:shipit:

**Pull request merged**